src/tUltimateLI.pas

```diff
@@ -1198,6 +1266,216 @@ procedure TuLI.SendLocoData(callByte: Byte; addr: Integer);
+    toSend := toSend + #$A + #$80 + #0 + #0;
+    if ((addr > 0) and (addr < _SLOTS_CNT)) then
+      for var i := 0 to _MAX_FUNC do
+        Self.sloty[addr].mausFunkce[i] := false;
+  end
+  else
+  begin
+    // Kennung (static)
+    toSend := toSend + #$50;
+
+    // F0 - F4
+    var tmp := (Byte(Self.sloty[addr].funkceType[0]) shl 4);
+    for var i := 1 to 4 do
+      tmp := tmp + (Byte(Self.sloty[addr].funkceType[i]) shl (i - 1));
+    toSend := toSend + AnsiChar(tmp);
+
+    // F5 - F12
+    tmp := 0;
+    for var i := 5 to 12 do
+      tmp := tmp + (Byte(Self.sloty[addr].funkceType[i]) shl (i - 5));
+    toSend := toSend + AnsiChar(tmp);
+
+    // R (emulated)
+    toSend := toSend + #$03; // 28 speed steps
+
+    Self.sloty[addr].mausFunkce := Self.sloty[addr].funkce;
+  end;
+
+  Self.WriteLog(tllCommands, 'PUT: function F13-F28 momentary information');
+  Self.Send(CreateBuf(toSend));
+end;
+
+/// /////////////////////////////////////////////////////////////////////////////
+
+procedure TuLI.SendLocoFunc13Momentary(callByte: Byte; addr: Integer);
+var
+  toSend: ShortString;
+  changed: boolean;
+begin
+  toSend := AnsiChar(callByte) + #$E4;
+  changed := false;
+
+  var addrOld := Self.FindSlot(callByte AND $1F);
+  if ((addrOld > -1) and (addrOld <> addr)) then
+  begin
+    // na mysi doslo ke zmene adresy z addrOld na addr
+    // -> odstranit slot addrOld
+    Self.sloty[addrOld].ReleaseLoko();
+    Self.sloty[addrOld].mausId := TSlot._MAUS_NULL;
+    changed := true;
+  end;
+
+  if (((addr >= 1) and (addr <= _SLOTS_CNT)) and (not Self.sloty[addr].isMaus))
+  then
+  begin
+    // obsazujeme slot adresou
+    Self.sloty[addr].mausId := (callByte AND $1F);
+    changed := true;
+  end;
+
+  if (changed) then
+  begin
+    Self.RepaintSlots(F_Slots);
+    TCPServer.BroadcastSlots();
+  end;
+
+  if ((addr = 0) or (addr > _SLOTS_CNT) or (not Self.sloty[addr].isLoko) or
+    (Self.sloty[addr].ukradeno)) then
+  begin
+    // lokomotiva neni rizena ovladacem
+    toSend := toSend + #$A + #$80 + #0 + #0;
+    if ((addr > 0) and (addr < _SLOTS_CNT)) then
+      for var i := 0 to _MAX_FUNC do
+        Self.sloty[addr].mausFunkce[i] := false;
+  end
+  else
+  begin
+    // Kennung (static)
+    toSend := toSend + #$51;
+
+    // F13 - F20
+    var tmp : Integer := 0;
+    for var i := 13 to 20 do
+      tmp := tmp + (Byte(Self.sloty[addr].funkceType[i]) shl (i - 13));
+    toSend := toSend + AnsiChar(tmp);
+
+    // F21 - F28
+    tmp := 0;
+    for var i := 21 to 28 do
+      tmp := tmp + (Byte(Self.sloty[addr].funkceType[i]) shl (i - 21));
+    toSend := toSend + AnsiChar(tmp);
+
+    // R (emulated)
+    toSend := toSend + #$03; // 28 speed steps
+
+    Self.sloty[addr].mausFunkce := Self.sloty[addr].funkce;
+  end;
+
+  Self.WriteLog(tllCommands, 'PUT: function F13-F28 momentary information');
+  Self.Send(CreateBuf(toSend));
+end;
+
+/// /////////////////////////////////////////////////////////////////////////////
+
```

Comment:
Nové odpovědi dle XpressNET specifikace 3.6.
Naprosto hloupě okopírovaný kód. Funguje, ale chtšlo by to to nějak výrazně zkrátit. 95% kódu se opakuje.
Kontrola, zda není jiná adresa loko a spol. je pořád stejné. Liší se jen 4 poslední byte, co se odesílají.